src/components/LayoutTree.tsx

```diff
@@ -364,6 +386,167 @@ const linearizeTree = (
   return thoughts
 }
 
+/** Renders a thought component for mapped treeThoughtsPositioned. */
+const RenderMappedThought = (props: TreeThoughtPositionedComponent) => {
+  const {
+    belowCursor,
+    cliff,
+    depth,
+    env,
+    height,
+    indexChild,
+    indexDescendant,
+    isCursor,
+    isTableCol1,
+    isTableCol2,
+    thoughtKey,
+    leaf,
+    path,
+    prevChild,
+    showContexts,
+    isLastVisible,
+    simplePath,
+    singleLineHeightWithCliff,
+    style,
+    thoughtId,
+    width,
+    autofocus,
+    x,
+    y: _y,
+    index,
+    viewportBottom,
+    treeThoughtsPositioned,
+    bulletWidth,
+    cursorUncleId,
+    setSize,
+    cliffPaddingStyle,
+    dragInProgress,
+    autofocusDepth,
+  } = props
+
+  const [y, setY] = useState(_y)
+
+  useEffect(() => {
+    if (y !== _y) {
```

Comment:
This check shouldn't be necessary, since `setY` won't actually trigger a rerender/update if `y === _y`, right?

Reply:
This check guarantees that it will update only and only if the previous value changes. This at the same time answers your concern about unnecessary re-render.